front/app/components/Form/index.tsx

```diff
@@ -69,21 +72,38 @@ const InvisibleSubmitButton = styled.button`
   visibility: hidden;
 `;
 
-const customAjv = createAjv({ useDefaults: true });
+type FormData = Record<string, any> | null | undefined;
+
+const customAjv = createAjv({ useDefaults: 'empty', removeAdditional: true });
 
 export const APIErrorsContext = React.createContext<CLErrors | undefined>(
   undefined
 );
-export const InputTermContext = React.createContext<InputTerm>('idea');
+
+export type ApiErrorGetter = (
+  errorKey: string,
+  fieldName: string
+) => Message | undefined;
+export type AjvErrorGetter = (
+  error: ErrorObject,
+  uischema?: UISchemaElement
+) => Message | undefined;
+
+export const FormContext = React.createContext<{
+  showAllErrors: boolean;
+  getApiErrorMessage: ApiErrorGetter;
+}>({ showAllErrors: false, getApiErrorMessage: () => undefined });
 
 interface Props {
-  schemaMultiloc: { [key in Locale]?: JsonSchema7 };
-  uiSchemaMultiloc: { [key in Locale]?: UISchemaElement };
+  schema: JsonSchema7;
+  uiSchema: UISchemaElement;
   onSubmit: (formData: FormData) => Promise<any>;
   initialFormData?: any;
   title?: ReactElement;
   submitOnEvent?: string;
-  onChange?: (FormData) => void;
+  getApiErrorMessage?: ApiErrorGetter;
+  getAjvErrorMessage: AjvErrorGetter;
+  onChange?: (formData: FormData) => void; // if you use this as a controlled form, you'll lose some extra validation and transformations as defined in the handleSubmit.
```

Comment:
Very small remark, but if you add such comments as TSDoc on the type instead, it gets picked up by IDEs and is easy to see while typing the prop value.

```suggestion
  /**
   * If you use this as a controlled form, you'll lose some extra validation and transformations as defined in the handleSubmit.
   */
  onChange?: (formData: FormData) => void;
```